### ISSUE TITLE
user activity detection inoperative on Windows (x64)

### ISSUE THREAD

**Issue by romw** [BOINC/boinc#1115]
**Reported by SilverKnight on 28 Dec 41748718 14:13 UTC**
The 6.12.34 build of Boinc has broken user activity detection (at least on the x64 version) - it is unknown weather this affects the x32 version

OS: Windows 7 x64
Inst: executable install downloaded from boinc.

6.10.58 functioned for detection.

Attempting to override project prefs.
Using the Boinc Manager - Preferences and setting "While computer is in use" to UNCHECKED and "Only after computer has been idle for 3.00 minutes", and with no restrictions on time, Boinc immediately starts processing, even though the system is "active".

On this system, Boinc is running as a service (as was the previously installed version)

Using the project prefs (downloaded for the project, e.g. clearing the local machine prefs) does not correct the issue - as soon as the "every day between the hours of" setting is reached, the project is spooled up and continues to run, irrespective of user activity, even though the while on batteries and while computer is in use are unchecked.

This may be related to the changes present for #463 (Linux version of this issue)

Feel free to contact me directly if I may be of assistance in testing (pre-built) patches.

**Comment by romw** [comment]
**Commented by Ageless on 26 Sep 41751781 10:13 UTC**
BOINC uses a separate program on Windows 7 and Vista to determine if the host computer is idle or not. This program is called '''boinctray.exe''' and it only starts up from the registry (when you reboot the system). 

When you do an upgrade installation of BOINC, all the executables in the BOINC programs directory will be a) unloaded from memory and b) deleted from the drive, before the installer installs the newer versions to that directory. So, doing an upgrade installation, you will find that boinctray.exe stops working. With that, the idle detection will stop working. 

Easily fixed, either reboot the system or navigate to your BOINC programs directory and (double)click on boinctray.exe, starting it. it won't show you that it's running in any way, other than through Windows task manager. 

Of course, you did also check that your Activity preference is to run Based on preferences? As "Run always" will also ignore any preferences you set. 

Please, in the future, before declaring this a bug immediately, come by on the [BOINC forums](http://boinc.berkeley.edu/dev/) and post about it. That way we can talk about it. 

I'm setting the ticket to 'worksforme', as in .33, .34, .37 and .38 the idle detection all works for me on Windows 7 64bit Ultimate.

**Comment by romw** [comment]
**Commented by JacobKlein on 5 Sep 41758275 20:26 UTC**
So, although the user is not prompted to reboot, they can expect their "while computer is in use" settings to not work when upgrading a service install?

That's really unfortunate, and it's a poor attitude for fixing the real problem. There must be a way to have the installer finish in a way that the preferences are working correctly, or the user must be instructed to launch a program or reboot.

The way it is now, the poor user is left thinking their preferences are being ignored. Is that what we want them to think?

**Comment by romw** [comment]
**Commented by Ageless on 21 Jan 41758637 14:13 UTC**
Replying to [JacobKlein](comment:2):
> The way it is now, the poor user is left thinking their preferences are being ignored. Is that what we want them to think?

See ticket #751.
It's really not the first time this came up.

**Comment by romw** [comment]
**Commented by JacobKlein on 17 Jan 41768985 01:20 UTC**
I never said that this issue wasn't new.[[BR]]My point was that it's misleading to close a ticket as "WorksForMe" when the issue is not resolved, and only a workaround exists to make the client function correctly.

If anything, this ticket #1142 should be marked as a duplicate of ticket #751, and then ticket #751 should be given enough attention to become actually fixed.

I have hoped for many months that ticket #751 will get fixed, but it hasn't (yet). I continue to hope.

**Comment by romw** [comment]
**Commented by Ageless on 27 May 41769753 15:06 UTC**
Duplicate of #751
